assets/ts/search.tsx

```diff
@@ -53,14 +54,50 @@ class Search {
         this.bindSearchForm();
     }
 
+    /// clean input keywords list
+    private reorganizeKeywords(keywords: string[]) {
+    let tmp: string[] = [];
+    /// do the search only on keywords length >= 2
+    for (let i=0; i < keywords.length; i++){
+      if (keywords[i].length === 1) keywords.splice(i,1);
+    }
+    /// Sort keywords from short to long
+    keywords.sort((a, b) => {
+        return b.length - a.length
+    });
+    /// remove the keywords contained in other keywords
+    for (let i = 0; i < keywords.length; i++) {
+      for (let j = 0; j < keywords.length; j++) {
+          if (i !== j) {
+            if (keywords[j].includes(keywords[i])) {
+              keywords.splice(i,1);
+              i=j=0;
+            }
+          }
+      }
+    }
+    return keywords;
+  }
```

Comment:
This seems to break a part of the functionality: try search `testest test`, and no result will be returned:

![image](https://user-images.githubusercontent.com/5889006/147682360-b012d425-195c-41ae-a36e-15edd60ed1be.png)

But it should return 2 pages:

![image](https://user-images.githubusercontent.com/5889006/147682416-515dd3a3-a818-4e09-aaf3-c778885a7e7d.png)

Since `test` is included in `testest`, `test` is removed, and regex will only match `testest`. But there's no page with such keyword.

----

Why is this function necessary? I don't understand it.

Reply:
Hi Jimmy,
The organizekeywords is a tentative to optimize the search including only the longest match, but I didn't realize this side effects.
I rewrite the code and now works fine, I also resolve a little bug that included the whitespace as search result.

Your idea is very similar to mine, but if you need to reinvent the wheel I think is better to try to integrate a 3rd party searching library.